playground/src/App.js

```diff
@@ -10,58 +9,109 @@ function validName(str) {
   else return false;
 }
 
+function encryptionKeyFilter(stateData, componentProps) {
+  // add filter here
+  return componentProps.encryptionKeys;
+}
+
+function composedNameCallback(stateData, componentProps) {
+  return `${stateData.name}-<random suffix>`;
+}
+
 function invalidCallback(stateData, componentProps) {
-  return !validName(stateData.name);
+  return !validName(stateData.name) || contains(["foo", "bar"], stateData.name);
 }
 
 function invalidTextCallback(stateData, componentProps) {
-  return !validName(stateData.name)
-    ? `Name ${stateData.name} is invalid.`
+  return contains(["foo", "bar"], stateData.name)
+    ? `Name ${stateData.name} already in use.`
     : `Invalid Name. Must match the regular expression: /^[A-z]([a-z0-9-]*[a-z0-9])?$/i`;
 }
 
+function invalidIopsCallback(stateData, componentProps) {
```

Comment:
remove, this is only for custom profiles. we currently are not supporting custom profiles. we went over this.

there should be three storage profiles:
- [ ] `3iops-tier`
- [ ] `5iops-tier`
- [ ] `10iops-tier`
https://cloud.ibm.com/docs/vpc?topic=vpc-block-storage-profiles&interface=ui